docs/UsingTheIOSubsystem.md

```diff
@@ -0,0 +1,615 @@
+are needed. This is purely a security measure. By default, enclaves have no
+access to files or sockets. The next section describes how to opt-in to
+various I/O features.
+
+Opting in
+---------
+
+An enclave application must first opt in to the I/O features it wishes to
+use. Opting in is a matter of (1) linking the desired module libraries and
+(2) calling functions to load the modules. I/O features are packaged as
+static libraries. This release provides the following modules.
+
+- **liboehostfs** -- access to non-secure host files and directories.
+- **liboehostsock** -- access to non-secure sockets.
+- **libhostresolver** -- access to network information.
+
+After linking modules, the enclave loads modules by calling one of the
+following.
+
+- **oe_load_module_host_file_system()**
+- **oe_load_module_host_socket_interface()**
+- **oe_load_module_host_resolver()**
+
+Operating system support
+------------------------
+
+The current version is Limited to Linux hosts but Windows host is under
+development now.
+
+File system path resolution
+---------------------------
+
+Path-oriented functions must resolve their path parameter to a file system
+device. There are two mechanisms for doing this. The first mechanisms employs
+the Linux **mount** function defined below.
+
+```
+#include <sys/mount.h>
+
+int mount(
+    const char* source,
+    const char* target,
+    const char* filesystemtype,
+    unsigned long mountflags,
+    const void* data);
+```
+
+This function attaches a file system to the directory specifies by **target**.
+**filesystemtype** parameter specifies the name of the file system. The next
+section will show how to mount the non-secure host file system.
+
+The second path-resolution mechanism employs the following functions.
+
+```
+oe_result_t oe_set_thread_devid(uint64_t devid);
+oe_result_t oe_clear_thread_devid();
+```
+
+The **oe_set_thread_devid** function sets the device id for the current
+thread. Once set, that device is used by path-oriented functions until
+it is cleared by calling **oe_clear_thread_devid**. These functions can be
+used to wrap calls to path-oriented functions. Consider the following function
+definition.
+
+```
+#include <stdio.h>
+#include <openenclave/enclave.h>
+
+FILE* fopen_by_device(uint64_t devid, const char *path, const char *mode)
+{
+    extern oe_result_t oe_set_thread_devid(uint64_t devid);
+    extern oe_result_t oe_clear_thread_devid(uint64_t devid);
+
+    oe_set_thread_devid(devid);
+    FILE* stream = fopen(path, mode);
+    oe_clear_thread_devid(devid);
+
+    return stream;
+}
+```
+
+This function can then be used to compose device-specific functions. For
+example.
+
+```
+FILE* fopen_host_file_system(const char *path, const char *mode)
+{
+    const uint64_t OE_DEVID_HOST_FILE_SYSTEM = 1;
+
+    return fopen_by_device(OE_DEVID_HOST_FILE_SYSTEM, path, mode);
+}
+```
+
+Caution: the thread-devid functions are experimental in the current release.
```

Comment:
Make this in Bold and/OR move this to the start of the thread-devid section i.e. line 99

Reply:
Ok. Resolved.